CHANGELOG.md

```diff
@@ -6,7 +6,8 @@
 
 ### Fixed
 * <How do the end-user experience this issue? what was the impact?> ([#????](https://github.com/realm/realm-core/issues/????), since v?.?.?)
-* None.
+* Fix data from a previous interrupted bootstrap was potentially being included with the bootstrap data during retry attempt
```

Comment:
Couldn't this potentially cause diverging history?

Reply:
Not immediately, but manually removing the extra entries may cause a diverging history error when merged with the server, since it expects those to not be in the local realm.

Reply:
>Not immediately, but manually removing the extra entries may cause a diverging history error

Not sure what you mean.

After thinking a bit more, I think it can actually cause orphaned objects not diverging history.

Reply:
Yes, orphaned objects is what I was trying to say

Reply:
@kmorkos correct me if i'm wrong, but I think this will just lead to compensating writes rather than diverging history.

But also, this changelog entry is super confusing. The audience for changelog entries are SDK engineers and end users who likely don't know about the pending bootstrap store. Perhaps something like

* If a sync session were interrupted by a disconnect while downloading a bootstrap more writes may have been made to the database than necessary when the sync session reconnected, and there may be objects stored that do not match the actual state of the server - potentially leading to compensating writes.

Also, I don't think this started in 14.8.0 - I think this started in v12.0.0 https://github.com/realm/realm-core/blob/master/CHANGELOG.md#1200-release-notes.

Reply:
I think the bug is that the client may end up with objects that the server doesn't think it has (if the object was in the new query's view during the first attempt, but then moved out that query's view before the second attempt).

From there, one of three things could happen:
* The object moves back into the client's query view at some point in the future, and we are _eventually consistent™️_ 
* The object never moves back into the client's query view, and the client just holds on to this stale view of an object it was never supposed to have indefinitely
* The client tries modifying the object at some point, at which point they'll get a compensating write because the server interprets it as modifying an object outside of their query view

I think for all intents and purposes @jbreams' description is more accurate than referring to them as "orphaned objects" unless that terminology is used elsewhere to refer to the above scenario

Reply:
I reworded the changelog entry based on Jonathan's recommendation - hopefully it is clearer now.